app/validators/fee/concerns/case_numbers_validator.rb

```diff
@@ -52,9 +54,14 @@ def validate_each_case_number
       end
 
       def validate_case_number(case_number)
-        add_error(:case_numbers, 'invalid') unless case_number.match?(CASE_NUMBER_PATTERN)
+        add_error(:case_numbers, 'invalid') unless case_number.match?(URN_PATTERN)
```

Comment:
can move this line `57` down to `validate_case_number_pattern` method

Reply:
I don't think I can. I am only calling validate_case_number_pattern where looks_like_a_case_number is true. But all case numbers (not just URNs) need to be validated against URN_PATTERN.